### PR TITLE
Fix ScrollIntoView - correct position for the focus

### PR DESCRIPTION
**Pull request by lacell75**
## Proposed changes

In case where there are several possible scrolls in the page(example: a page with a left menu), the scroll doesn't work on the other parts of the page because the focus by default is positioned at x:0 , y:0.
To fix it, I just added the origin element in argument.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # 

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Comment by christian-bromann** [comment]
@lacell75 thanks for raising the PR

The solution seems reasonable even though I am not sure if it has implications on other use cases. However the test seems to legitimately fail: `e2e/wdio/headless/test.e2e.ts:91:36`

**Comment by lacell75** [comment]
Yes. The calculation for the start, end,.. position is different if a
scroll has been done before.
For now, I didn't find the solution to fix it.

File: e2e/wdio/headless/test.e2e.ts [hunk 1]
```diff
@@ -100,15 +100,26 @@ describe('main suite 1', () => {
...
+            await browser.switchToWindow(handles[0])
+            await browser.pause(500)
+            searchInput = await $('.searchinput')
             await searchInput.scrollIntoView(params)
             await browser.pause(500)
             const [wdioX, wdioY] = await browser.execute(() => [
                 window.scrollX, window.scrollY
             ])
 
+            await browser.switchToWindow(handles[1])
```
Comment (christian-bromann) [inline e2e/wdio/headless/test.e2e.ts:119]: Instead of working with different window handles, can we just reload the page to reset the scroll position?

**Comment by lacell75** [comment]
@christian-bromann PR ready to review

**Comment by christian-bromann** [comment]
> PR ready to review

@lacell75 did you see my comment? I wonder if we can optimize the e2e test a bit

**Comment by lacell75** [comment]
> > PR ready to review
> 
> @lacell75 did you see my comment? I wonder if we can optimize the e2e test a bit

Sorry, I had not seen your comment. Improvement done

File: e2e/wdio/headless/test.e2e.ts [hunk 2]
```diff
@@ -100,15 +100,22 @@ describe('main suite 1', () => {
     it('should be able to handle successive scrollIntoView', async () => {
         await browser.url('http://guinea-pig.webdriver.io')
         await browser.setWindowSize(500, 500)
-        const searchInput = await $('.searchinput')
+        let searchInput = await $('.searchinput')
```
Comment (christian-bromann) [inline e2e/wdio/headless/test.e2e.ts:103]:
```suggestion
```

File: e2e/wdio/headless/test.e2e.ts [hunk 3]
```diff
@@ -100,15 +100,22 @@ describe('main suite 1', () => {
...
 
         const scrollAndCheck = async (params?: ScrollIntoViewOptions | boolean) => {
+
+            await browser.refresh()
+            searchInput = await $('.searchinput')
```
Comment (christian-bromann) [inline e2e/wdio/headless/test.e2e.ts:108]:
```suggestion
            const searchInput = await $('.searchinput')
```

File: e2e/wdio/headless/test.e2e.ts [hunk 4]
```diff
@@ -100,15 +100,22 @@ describe('main suite 1', () => {
...
+            await browser.pause(500)
             await searchInput.scrollIntoView(params)
             await browser.pause(500)
             const [wdioX, wdioY] = await browser.execute(() => [
                 window.scrollX, window.scrollY
             ])
 
+            await browser.refresh()
+            searchInput = await $('.searchinput')
```
Comment (christian-bromann) [inline e2e/wdio/headless/test.e2e.ts:117]: WebdriverIO should be able to re-fetch stale elements

```suggestion
```

Comment (lacell75) [reply e2e/wdio/headless/test.e2e.ts:117]: I have this error : `stale element reference: The element with reference ELEMENT-14 is stale; either the element is no longer attached to the DOM, it is not in the current frame context, or the document has been refreshed`

Comment (lacell75) [reply e2e/wdio/headless/test.e2e.ts]: I reverted my changes

**Comment by lacell75** [comment]
I think everything is ok

**Review by christian-bromann — APPROVED** [review]
Great work, amazing change 👍

**Pull request merged**